new-log-viewer/src/services/LogExportManager.ts

```diff
@@ -0,0 +1,42 @@
+import downloadDecompressedLogs from "../utils/downloadDecompressedLogs";
+
+
+class LogExportManager {
+    private blob: Blob;
+
+    private numChunks: number;
+
+    constructor (numChunks?: number) {
+        this.blob = new Blob();
+        this.numChunks = numChunks ?? 0;
+    }
+
+    getBlob (): Blob {
+        return this.blob;
+    }
+
+    appendChunkData (chunkData: string) {
+        this.blob = new Blob([this.blob,
+            chunkData], {type: "text/plain"});
+    }
+
+    getNumChunks (): number {
+        return this.numChunks;
+    }
+
+    setNumChunks (numChunks: number) {
+        this.numChunks = numChunks;
+    }
```

Comment:
Let's see the comments at the `reset()` method first. Maybe we do not need to change the `numChunk` dynamically.